backend/services/wasabi_service.py

```
@@ -0,0 +1,667 @@
+                    'Key': s3_key,
+                    'Body': file_bytes,
+                    'ContentType': content_type
+                }
+                # Only add ACL if public_read is enabled and ACLs haven't been found to be unsupported
+                if self.public_read and not acl_not_supported:
+                    put_params['ACL'] = 'public-read'
+                
+                # Wrap blocking boto3 call with executor
+                await asyncio.get_running_loop().run_in_executor(
+                    None,
+                    lambda: self.s3_client.put_object(**put_params)
```

Ruff (0.14.3)

Function definition does not bind loop variable `put_params`

(B023)